### PR TITLE
[flutter_tools] support trailing args

### LINKED ISSUE #85891: Arguments passed to "flutter test" after the first script name are interpreted as filenames
I'm not sure if this is a bug, but it's not consistent with the other tools so I thought it was worth a discussion.

Running tests using `pub` or `dart run test:test`, arguments can come before or after the script name. These all work as expected (using the JSON reporter):

```
dart run test:test -r json test/foo_test.dart
dart run test:test test/foo_test.dart -r json
pub run test -r json test/foo_test.dart
pub run test test/foo_test.dart -r json
flutter test -r json test/foo_test.dart
```

However putting args after the script for `flutter test` fails because `-r` and `json` are interpreted as script filenames:

```
flutter test test/foo_test.dart -r json

00:00 +0 -1: loading /Users/danny/Desktop/flutter_application_1/-r [E]                                                                                         
  Failed to load "/Users/danny/Desktop/flutter_application_1/-r": Does not exist.
00:00 +0 -2: loading /Users/danny/Desktop/flutter_application_1/json [E]                                                                                       
  Failed to load "/Users/danny/Desktop/flutter_application_1/json": Does not exist.
00:02 +1 -2: Some tests failed. 
```

This came up because in the latest version of the VS Code extension I tidied up handling of args to be more consistent, so there are `"toolArgs"` that always go before the script and `"args"` always go after. However some users already had `"args"` being used in test projects.

As an immediate fix, I'm going to insert `"args"` also before the script name for `flutter test`, however it would be nice if this worked the same as the other tools to avoid this inconsistency to simplify things for users when setting `"args"`/`"toolArgs"`.

@jonahwilliams what do you think? I had a quick look at the code but it wasn't obvious to me why this doesn't work (they're read from `args.rest`, but as far as I could tell that was the same for `package:test` so I'm not sure why it's working different there).

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/85891

Allow all flutter commands to support trailing arguments.

### CODE REVIEW

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -159,8 +159,7 @@ abstract class FlutterCommand extends Command<void> {
   @override
   ArgParser get argParser => _argParser;
   final ArgParser _argParser = ArgParser(
-    allowTrailingOptions: false,
-    usageLineLength: globals.outputPreferences.wrapText ? globals.outputPreferences.wrapColumn : null,
+    usageLineLength: globals.outputPreferences.wrapText? globals.outputPreferences.wrapColumn : null,
```
Comment (DanTup) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:162]: Was the removal of this space accidental?

**Review by DanTup — APPROVED** [review]

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (DanTup) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:162]: (and Thanks! 😄)

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart]: oops!

**Comment by jonahwilliams** [comment]
Looks like this does break how we pass things through to pub. 😓

**Comment by DanTup** [comment]
heh, I thought it seemed to easy 😄

If you don't have time to debug it and would prefer I had a look, let me know. It'd definitely be nice for me if we could make this consistent (esp. as some of those code will move from Dart-Code to the SDKs with DAP :-))

**Comment by DanTup** [comment]
LGTM, thank you! 🙂

**Pull request merged**